File: P07/P50497_en/main.java
import java.util.Scanner;

class Main {

  public static void main(String[] args) {
    final Scanner sc = new Scanner(System.in);
    while (sc.hasNext()) {
      String s = sc.next();
      System.out.println(Solution.isPalindrome(s) ? "1" : "0");
      int n = s.length();
      if (n > 100) {
        StringBuilder str = new StringBuilder(s);
        char c = s.charAt(0);
        str.setCharAt(0, 'a');
        String sTmp = str.toString();
        for (int rep = 1000; rep > 0; --rep) System.out.println(Solution.isPalindrome(sTmp) ? "1" : "0");
        str.setCharAt(0, c);
        c = s.charAt(1);
        str.setCharAt(1, 'a');
        sTmp = str.toString();
        for (int rep = 1000; rep > 0; --rep) System.out.println(Solution.isPalindrome(sTmp) ? "1" : "0");
        str.setCharAt(1, c);
        c = s.charAt(3);
        str.setCharAt(3, 'a');
        sTmp = str.toString();
        for (int rep = 1000; rep > 0; --rep) System.out.println(Solution.isPalindrome(sTmp) ? "1" : "0");
        str.setCharAt(3, c);
        c = s.charAt(n - 1);
        str.setCharAt(n - 1, 'a');
        sTmp = str.toString();
        for (int rep = 1000; rep > 0; --rep) System.out.println(Solution.isPalindrome(sTmp) ? "1" : "0");
        str.setCharAt(n - 1, c);
        c = s.charAt(n - 2);
        str.setCharAt(n - 2, 'a');
        sTmp = str.toString();
        for (int rep = 1000; rep > 0; --rep) System.out.println(Solution.isPalindrome(sTmp) ? "1" : "0");
        str.setCharAt(n - 2, c);
        c = s.charAt(n - 7);
        str.setCharAt(n - 7, 'a');
        sTmp = str.toString();
        for (int rep = 1000; rep > 0; --rep) System.out.println(Solution.isPalindrome(sTmp) ? "1" : "0");
        str.setCharAt(n - 7, c);
      }
    }
  }

}

File: P07/P50497_en/main.c
#include <stdio.h>
#include <string.h>

int is_palindrome(const char s[], int ssize);

int main() {
  int n, rep;
	char s[2000010], c;
	while (scanf("%s", s) > 0) {
    n = (int)strlen(s);
    printf("%d\n", is_palindrome(s, n));
    if (n > 100) {
      c = s[0];
      s[0] = 'a';
      for (rep = 1000; rep > 0; --rep) printf("%d\n", is_palindrome(s, n));
      s[0] = c;
      c = s[1];
      s[1] = 'a';
      for (rep = 1000; rep > 0; --rep) printf("%d\n", is_palindrome(s, n));
      s[1] = c;
      c = s[3];
      s[3] = 'a';
      for (rep = 1000; rep > 0; --rep) printf("%d\n", is_palindrome(s, n));
      s[3] = c;
      c = s[n - 1];
      s[n - 1] = 'a';
      for (rep = 1000; rep > 0; --rep) printf("%d\n", is_palindrome(s, n));
      s[n - 1] = c;
      c = s[n - 2];
      s[n - 2] = 'a';
      for (rep = 1000; rep > 0; --rep) printf("%d\n", is_palindrome(s, n));
      s[n - 2] = c;
      c = s[n - 7];
      s[n - 7] = 'a';
      for (rep = 1000; rep > 0; --rep) printf("%d\n", is_palindrome(s, n));
      s[n - 7] = c;
    } 
	}
	return 0;
}
